genie-web/src/main/resources/static/scripts/components/JobDetails.js

```diff
@@ -46,9 +51,19 @@ export default class JobDetails extends React.Component {
     const url = row._links.self.href;
     fetch(url).done(job => {
       this.setState({ job });
-      fetch(job._links.command.href)
-        .done(command => {
-          this.setState({ command });
+
+      $.when(
+        fetch(job._links.cluster.href),
+        fetch(job._links.command.href),
+        fetch(job._links.applications.href)
+      )
+        .done((clusters, commands, applications) => {
+          console.log(applications);
+          this.setState({
+            cluster: clusters[0],
+            command: commands[0],
+            applications: applications[0]
```

Comment:
cluster and command are a single entity, but application is an array.
Seems strange they get all treated the same here.

Reply:
@mprimi This is specific to `jquery`'s `when` interface. The `applications[0]` returns an array hence its assgined to the key `applications`. Where as `commands[0]` returns an object and hence the key is named `command`.